ui/app/adapters/job.js

```diff
@@ -168,11 +168,20 @@ export default class JobAdapter extends WatchableNamespaceIDs {
     });
   }
 
-  runAction(job, action, allocID) {
-    let messageBuffer = '';
-
+  /**
+   *
+   * @param {import('../models/job').default} job
+   * @param {import('../models/action').default} action
+   * @param {string} allocID
+   * @returns {string}
+   */
+  getActionSocketUrl(job, action, allocID) {
```

Comment:
This file previously did much more than an Ember adapter ought to; it has since been moved to a dedicated Action service.